### PR TITLE
Refactoring and dependencies change in gobblin-compliance

### PR DESCRIPTION
**Pull request by aditya1105**

### CODE REVIEW

File: gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/utils/DatasetUtils.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import gobblin.compliance.HivePartitionDataset;
+import gobblin.data.management.copy.hive.HiveDataset;
+
+
+/**
+ * @author adsharma
+ */
+@Slf4j
+public class DatasetUtils {
```
Comment (ibuenros) [inline gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/utils/DatasetUtils.java]: Can you add javadoc here?

File: gobblin-modules/gobblin-azkaban/build.gradle [hunk 2]
```diff
@@ -21,6 +21,7 @@ dependencies {
   compile project(":gobblin-api")
   compile project(":gobblin-compaction")
   compile project(":gobblin-data-management")
+  compile project(":gobblin-modules:gobblin-compliance")
```
Comment (ibuenros) [inline gobblin-modules/gobblin-azkaban/build.gradle]: Why is this needed?

File: gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/utils/DatasetUtils.java (hunk 1 above)
Comment (aditya1105) [inline gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/utils/DatasetUtils.java]: Added javadoc for each method

File: gobblin-modules/gobblin-azkaban/build.gradle (hunk 2 above)
Comment (aditya1105) [inline gobblin-modules/gobblin-azkaban/build.gradle]: This is for my next pr which will use some classes from compliance.

Comment (aditya1105) [inline gobblin-modules/gobblin-azkaban/build.gradle]: I have removed this dependency. Will add it with next pr only
